File: quotebot-python/quotebot/automatic_quote_bot.py
from quotebot.ad_space import AdSpace
from quotebot.blog_auction_task import BlogAuctionTask


class AutomaticQuoteBot:

    def send_all_quotes(self, mode):
        blogs = AdSpace.get_ad_spaces()
        for blog in blogs:
            auction_task = BlogAuctionTask()
            auction_task.price_and_publish(blog, mode)

File: quotebot-elixir/lib/automatic_quote_bot.ex
defmodule Quotebot.AutomaticQuoteBot do
  @moduledoc """
  AutomaticQuoteBot sends quotes to all ad spaces.
  Hard-coded dependencies make this difficult to test.
  """

  defstruct []

  def new do
    %__MODULE__{}
  end

  def send_all_quotes(_bot, mode) do
    blogs = Quotebot.AdSpace.get_ad_spaces()

    Enum.each(blogs, fn blog ->
      auction_task = Quotebot.BlogAuctionTask.new()
      Quotebot.BlogAuctionTask.price_and_publish(auction_task, blog, mode)
    end)
  end
end
